src/three-components/ModelScene.js

```diff
@@ -282,8 +273,9 @@ export default class ModelScene extends Scene {
    * Called to update the shadow rendering when the room or model changes.
    */
   updateStaticShadow() {
+    this.model.remove(this.shadow);
```

Comment:
FWIW `remove` can have unanticipated side-effects (see https://github.com/mrdoob/three.js/blob/master/src/core/Object3D.js#L423). It seems more hygienic only to remove it under the appropriate condition.

Reply:
:+1:

Reply:
Yeah. It is better to do `this.shadow.visible = false;`.

Reply:
TIL, thanks all!